runtime/integration-tests/src/generic/environment.rs

```diff
@@ -26,11 +31,75 @@ pub trait Env<T: Runtime> {
 	/// Load the environment from a storage
 	fn from_storage(storage: Storage) -> Self;
 
+	/// Creates an extrinsic, used by mainly by the own environment.
+	/// To create and submit an extrinsic, see `submit()`
+	fn create_extrinsic(
+		&self,
+		who: Keyring,
+		call: impl Into<T::RuntimeCall>,
+	) -> <T::Block as Block>::Extrinsic {
+		self.state(|| {
+			let runtime_call = call.into();
+			let nonce = frame_system::Pallet::<T>::account(who.to_account_id()).nonce;
```

Comment:
Seems like extracting the expected `nonce` directly from `frame_system` allows to not manage an `nonce` in the `fudge::companion` struct

Reply:
*This only works if the extrinsic is applied instantly*